File: frontend/src/interface/booking.ts
export interface bookingInterface {
  ID?: number; // Booking.ID
  user_id: number;
  showdate_id: number;
  zone_id: number;
  queue_number: number; // โซนยืนจะถูกคำนวณใน backend
  total_price: number;
  booking_status_id: number; // 1=pending, 2=paid, 3=expired (ตามระบบคุณ)
  booking_date: string; // ISO datetime
  expired_date: string;
  seat_ids?: number[]; // รายการ SeatAvailable.ID ที่จอง (ถ้ามี)
}

File: backend/entity/booking.go
package entity

import (
	"time"
	"gorm.io/gorm"
)

type Booking struct {
	gorm.Model
	UserID uint `json:"user_id"`
	User *User `gorm:"foreignKey:UserID" json:"user"`
	ShowDateID uint `json:"showdate_id"`
	ShowDate *ShowDate `gorm:"foreignKey:ShowDateID" json:"showdate"`
	ZoneID uint `json:"zone_id"`
	Zone *Zone `gorm:"foreignKey:ZoneID" json:"zone"`
	QueueNumber int `json:"queue_number"`
	TotalPrice int `json:"total_price"`
	BookingStatusID uint `json:"booking_status_id"`
	BookingStatus *BookingStatus `gorm:"foreignKey:BookingStatusID" json:"booking_status"`
	BookingDate time.Time `json:"booking_date"`
	ExpiredDate time.Time `json:"expired_date"`
}